datatree/datatree.py

```diff
@@ -1345,3 +1345,223 @@ def to_zarr(
+        ----------
+        **kwargs : dict
+            Additional keyword arguments passed to ``dask.persist``.
+
+        See Also
+        --------
+        dask.persist
+        """
+        new_datatree_dict = {
+            node.path: node.ds.persist(**kwargs) for node in self.subtree
+        }
+        return DataTree.from_dict(new_datatree_dict)
+
+    def __dask_tokenize__(self):
+        from dask.base import normalize_token
+
+        # This method should return a value fully representative of the object
+        # here. ``xarray.Dataset`` implements a method that accomplishes this,
+        # and ``DataTree`` is just fundamentally defining relationships between
+        # these ``Dataset``s. So here we re-use the ``Dataset`` tokenization and
+        # incorporate the ancestry as an additional component (encoded in the
+        # incorporate the ancestry as an additional component (encoded in the
+        # names of the datasets).
+
+        ds_tokens = {node.path: node.ds.__dask_tokenize__() for node in self.subtree}
+        ds_tokens = {node.path: node.ds.__dask_tokenize__() for node in self.subtree}
+
+        return normalize_token((type(self), ds_tokens))
+
+        return normalize_token((type(self), ds_tokens))
+
+    def __dask_graph__(self):
+        graphs = {node.path: node.ds.__dask_graph__() for node in self.subtree}
+        graphs = {node.path: node.ds.__dask_graph__() for node in self.subtree}
+        graphs = {k: v for k, v in graphs.items() if v is not None}
+
+        if not graphs:
+            return None
+        else:
+            try:
+                from dask.highlevelgraph import HighLevelGraph
+
+                return HighLevelGraph.merge(*graphs.values())
+            except ImportError:
+                from dask import sharedict
+
+                return sharedict.merge(*graphs.values())
+
+    def __dask_keys__(self):
+        return [node.ds.__dask_keys__() for node in self.subtree]
+        return [node.ds.__dask_keys__() for node in self.subtree]
+
+    def __dask_layers__(self):
+        all_keys = self.__dask_keys__()
+        return sum((all_keys), ())
+
+    @property
+    def __dask_optimize__(self):
+        import dask.array as da
+
+        return da.Array.__dask_optimize__
+
+    @property
+    def __dask_scheduler__(self):
+        import dask.array as da
+
+        return da.Array.__dask_scheduler__
+
+    def __dask_postcompute__(self):
+        return self._dask_postcompute, ()
+
+    def _dask_postcompute(self: DataTree, results: Iterable[DatasetView]) -> DataTree:
+        from dask import is_dask_collection
+
+        datatree_nodes = {}
+        results_iter = iter(results)
+
+        for node in self.subtree:
+            if is_dask_collection(node.ds):
+                finalize, args = node.ds.__dask_postcompute__()
+                # darothen: Are we sure that results_iter is ordered the same as
+                # self.subtree?
+                # self.subtree?
```

Comment:
Where does the iterable of `DatasetView` come from? Presumably you are looping over the nodes, but where are you doing that? Like I don't understand where `results` is passed in from.